sys/posix/sockets/posix_sockets.c

```diff
@@ -792,8 +795,12 @@ ssize_t recvfrom(int socket, void *restrict buffer, size_t length, int flags,
     switch (s->type) {
 #ifdef MODULE_SOCK_IP
         case SOCK_RAW:
-            /* TODO: apply configured timeout */
-            if ((res = sock_ip_recv(&s->sock->raw, buffer, length, SOCK_NO_TIMEOUT,
+            if ((res = sock_ip_recv(&s->sock->raw, buffer, length,
+#ifdef POSIX_SETSOCKOPT
+                                s->recv_timeout,
+#else
+                                SOCK_NO_TIMEOUT,
+#endif
```

Comment:
I would prefer to just set a const variable above in the ifdef and use it here and below. Same goes for accept.